protocols/v2/subprotocols/mining/src/lib.rs

```diff
@@ -336,6 +378,31 @@ impl ExtendedExtranonce {
         }
     }
 
+    /// Suppose that P receives an Extranonce type from the an upstream. Then range_0 (that should
```

Comment:
Refrase with something more like:

Used by downstream when an Extranonce received from upstream needs to be converted into and ExtendedExtranonce (eg when an extended channel is opened) .... If the byte reserved to the receiver (range1 and range2) are not set to 0 return None otherwise return Some(ExtendedExtranonce)